test/studies/champs/functions.bash

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env bash
+
+# Helper functions for logging and marking test start/stop. Formats output in
+# the way the testing infrastructure expects.
+
+function subtest_start() {
+  subtest_start_time=$(date +%s)
+  echo "[Starting subtest - $(date)]"
+}
+
+function subtest_end() {
+  local subtest_end_time=$(date +%s)
+  local elapsed=$(( $subtest_end_time - $subtest_start_time ))
+  echo "[Finished subtest \"studies/champs\" - $elapsed seconds"
+}
+
+function test_start() {
+  subtest_test_name=$1
+  echo "[test: \"$subtest_test_name\"]"
+  SECONDS=0
+}
+
+function test_end() {
+  elapsed=$SECONDS
+  echo "[Elapsed time to compile and execute all versions of \"$subtest_test_name\" - $elapsed.000 seconds]"
+  unset subtest_test_name
+}
+
+function log_error() {
+  local msg=$@
+  echo "[Error ${msg}]"
+}
+
+function log_fatal_error() {
+  local msg=$@
+  echo "[Error ${msg}]"
+  if [[ -n $subtest_test_name ]]; then
+    test_end
+  fi
+  subtest_end
+  exit 0
+}
+
+function log_success() {
+  local msg=$@
+  echo "[Success matching ${msg}]"
+}
+
+function apply_patch() {
+  local patch_file=$@
+  echo "Applying patch ${patch_file}"
+  if ! patch -p1 < $patch_file ; then
+    log_fatal_error "applying patch"
+  else
+    echo "applied patch successfully"
+  fi
+}
+
+function test_compile_exec() {
```

Comment:
Reminder that `chpl` is not in `PATH` for nightly testing, and presumably champs make targets call `chpl`. Arkouda sub_test does:


```sh
# Arkouda needs chpl in PATH
bin_subdir=$($CHPL_HOME/util/chplenv/chpl_bin_subdir.py)
export "PATH=$CHPL_HOME/bin/$bin_subdir:$PATH"
chpl --version
```